io-sim/src/Control/Monad/IOSim/Internal.hs

```diff
@@ -66,12 +66,18 @@ import qualified Data.Set as Set
 import           Data.Time (UTCTime (..), fromGregorian)
 import           Data.Dynamic
 
-import           Control.Exception (assert)
-import           Control.Monad (join)
+import           Control.Applicative (Alternative (..), liftA2)
+import           Control.Exception (ErrorCall (..), NonTermination (..),
+                   assert, asyncExceptionFromException,
+                   asyncExceptionToException, throw)
+import           Control.Monad (MonadPlus, join)
+import           Control.Monad.Fix (MonadFix (..))
+import qualified System.IO.Error as IO.Error (userError)
```

Comment:
Unused import?

Reply:
It is used by `MonadFail` instance, [ref](https://github.com/input-output-hk/ouroboros-network/blob/master/io-sim/src/Control/Monad/IOSim/Types.hs?plain=1#L239).